recipes/sentry-crashpad/all/conanfile.py

```diff
@@ -0,0 +1,126 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.28.0"
+
+
+class SentryCrashpadConan(ConanFile):
+    name = "sentry-crashpad"
+    description = "Crashpad is a crash-reporting system."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/getsentry/sentry-native"
+    license = "Apache-2.0"
+    topics = ("conan", "crashpad", "error-reporting", "crash-reporting")
+    provides = "crashpad", "mini_chromium"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+        "with_tls": ["openssl", False],
+    }
+    default_options = {
+        "fPIC": True,
+        "with_tls": "openssl",
+    }
+    exports_sources = "CMakeLists.txt", "patches/*"
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+        if self.options.with_tls:
```

Comment:
Why do you condition `openssl` requirement with an option?
According to the sources it will always look for it on `Linux` and `Android`. It could end up picking up the system instal of if this option is set to false.

Reply:
Fixed it.
By patching the cmake script, it can be kept as an option.

Reply:
Why changing the default behavior with an option?

Reply:
because default behavior is bad, unpredictable.

Reply:
Strong argument haha

Reply:
Indeed, that's the reason.

I think openssl is only used by the `crashpad_handler` executable. Not by the library.